src/Models/InventoryUpdateResult.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @copyright 2019 Wayfair LLC - All rights reserved
+ */
+
+namespace Wayfair\Models;
+
+use Plenty\Modules\Plugin\DataBase\Contracts\Model;
+
+/**
+ * Class InventoryUpdateResult
+ *
+ * @package Wayfair\Models
+ */
+class InventoryUpdateResult
+{
+    public function __construct(
+        bool $fullInventory = false,
+        int $dtosAttempted = 0,
+        int $variationsAttempted = 0,
+        int $dtosSaved = 0,
+        int $dtosFailed = 0,
+        int $elapsedTime = 0
+    ) {
+        $this->fullInventory = $fullInventory;
+        $this->dtosAttempted = $dtosAttempted;
+        $this->variationsAttempted = $variationsAttempted;
+        $this->dtosSaved = $dtosSaved;
+        $this->dtosFailed = $dtosFailed;
+        $this->elapsedTime = $elapsedTime;
+    }
+
+    /** @var bool */
+    public $fullInventory;
+
+    /** @var int */
+    public $dtosAttempted;
+
+    /** @var int */
+    public $variationsAttempted;
+
+    /** @var int */
+    public $dtosSaved;
+
+    /** @var int */
+    public $dtosFailed;
+
+    /** @var int */
+    public $elapsedTime;
+
+    public function toArray(): array
+    {
+        return [
+            'fullInventory' =>  $this->fullInventory,
+            'dtosAttempted' => $this->dtosAttempted,
+            'dtosSaved' => $this->dtosSaved,
+            'dtosFailed' => $this->dtosFailed,
+            'elapsedTime' => $this->elapsedTime,
+            'variationsAttempted' => $this->variationsAttempted
+        ];
+    }
+
+    public function isSuccessful()
+    {
+        return $this->variationsAttempted == 0 xor $this->dtosFailed == 0;

```

Comment:
For a partial sync isn't it possible that no variations were attempted and therefore no DTOs failed and then this would be a successful update result? Also, in what scenario would the variationsAttempted be 0 but the dtosFailed is more than 0 and why would this be considered successful?

Reply:
I think you're right - this should just be an `or`, not an `xor`.